nrich-search/README.md

```diff
@@ -150,76 +176,222 @@ public class CarSearchResult {
+[`StringSearchExecutor`][string-search-executor-url] and [`NaturalIdSearchExecturor`][natural-id-search-executor-url] in user defined JPA repository.
+
+An example of such a repository would be:
+
+```java
+public interface CarSearchDemoRepository extends JpaRepository<Car, Long>, SearchExecutor<Car>, StringSearchExecutor<Car>, NaturalIdSearchExecutor<Car> {
 
+}
 ```
 
-    SubqueryConfiguration subqueryConfiguration = SubqueryConfiguration.builder()
-                .rootEntity(UserRole.class)
-                .propertyPrefix("userRole")
-                .joinBy(new SearchPropertyJoin("id", "user.id")).build();
+## SearchConfiguration
+
+[`SearchConfiguration`][search-configuration-url] has a lot of options that let users customize query creation and result resolution.
+
+##### Defining result class
+
+If a regular class needs to be returned instead of projection,`.resultClass(SearchCarResult.class)` can be omitted and return class will be resolved from repository type parameter.
+Instead of a result class users can also manually define a list of projections using `projectionList` property.
+
+#### Defining table joins
+
+Users can define joins (fetches) (so no data is fetched in one select) by using `joinList`.
+Supported joins are **inner joins** and **left outer joins**.
+Joins are supplied by using the class `SearchJoin`.
+
+#### Defining property mapping
+
+`resolvePropertyMappingUsingPrefix` resolves properties by prefix, but it is also possible to write explicit property mapping from search request to searched entity by using`propertyMappingList`.
+
+#### Default operators
+
+Default operators (see [`DefaultSearchOperator`][default-search-operator-url]) are for String _**ILIKE**_ (uses `criteriaBuilder.like` with lower call before), for range search _**GT**_ (
+uses `criteriaBuilder.greaterThan`), _**LT**_ (uses `criteriaBuilder.lessThan`), _**GE**_ (uses `criteriaBuilder.greaterThanOrEqualTo`), _**LE**_ (uses `criteriaBuilder.lessThanOrEqualTo`)
+and for all other classes _**EQ**_ but this can be overridden either on type level or on property level by using `searchOperatorOverrideList`.
+
+Queries on plural associations are done using exists query (to avoid duplicate results) but that can be overridden by property `pluralAssociationRestrictionType`.
+Additional restrictions (not dependent on data in search request, for example security restrictions) can be specified using `additionalRestrictionResolverList`.
+
+#### Setting **and** / **or** operator
+
+[`SearchConfiguration`][search-configuration-url] supports matching **all** values (default behaviour), operator and is used for query predicate creation.
+[`SearchConfiguration`][search-configuration-url] also supports matching **any** value by setting `anyMatch` parameter to `true`, then or operator is used when creating query predicate.
```

Comment:
While reading this, I'm not sure if we maybe need to change the default behavior. Maybe **or** operator should be the default here. What do you think?

Reply:
I think that this should stay the same because I believe that the use of conjunction is a bit more prevalent while building applications, such in the case of filters, which are very commonly used.

Reply:
Yes, that is true. I was probably thinking about a quick search when I wrote this comment. In such a case or operator as default is maybe more convenient. However, I agree about filters and everything you said so we should leave it like this.